### PR TITLE
[cli] Add repo linking support for `deploy` command

### PR DESCRIPTION
**Pull request by TooTallNate**
Adds support for `vercel deploy` command when the repository has been linked via `vercel link --repo`.

### CODE REVIEW

File: packages/cli/src/commands/deploy/index.ts [hunk 1]
```diff
@@ -357,6 +345,12 @@ export default async (client: Client): Promise<number> => {
     }
   }
 
+  // For repo-style linking, reset the path to the root of the repository
+  //console.log(link);
+  if (link.status === 'linked' && link.repoRoot) {
+    cwd = join(link.repoRoot);
```
Comment (cb1kenobi) [inline packages/cli/src/commands/deploy/index.ts]: I don't think the `join()` is doing anything. Any truthy value for `repoRoot` will be just end up being returned by `join()`. Did you mean `resolve()`?

File: packages/cli/src/util/deploy/process-deployment.ts [hunk 2]
```diff
@@ -72,10 +70,9 @@ export default async function processDeployment({
     rootDirectory,
   } = args;
 
-  const { debug } = output;
-
+  const client = now._client;
```
Comment (cb1kenobi) [inline packages/cli/src/util/deploy/process-deployment.ts:73]: Can remove the `_` and make `client` a public property?

File: packages/cli/src/util/link/repo.ts [hunk 3]
```diff
@@ -270,3 +282,13 @@ export function findProjectFromPath(
       );
     });
 }
+
+/**
+ * TODO: remove
```
Comment (cb1kenobi) [inline packages/cli/src/util/link/repo.ts:287]: Out of curiosity, remove when and why?

File: packages/cli/src/util/projects/get-project-by-cwd-or-link.ts [hunk 4]
```diff
@@ -26,27 +25,10 @@ export default async function getProjectByCwdOrLink({
...
-  }
-
   // ensure the current directory is a linked project
-  const linkedProject = await ensureLink(
-    commandName,
-    client,
-    pathValidation.path,
-    { autoConfirm }
-  );
+  const linkedProject = await ensureLink(commandName, client, cwd, {
```
Comment (cb1kenobi) [inline packages/cli/src/util/projects/get-project-by-cwd-or-link.ts:29]: **nitpick (non-blocking):** ‏I think the old formatting with params on separate lines is easier to read that the changed code.

File: packages/cli/src/util/validate-paths.ts [hunk 5]
```diff
@@ -1,14 +1,11 @@
-import { lstat as lstatRaw } from 'fs';
-import { promisify } from 'util';
+import { lstat } from 'fs-extra';
```
Comment (cb1kenobi) [inline packages/cli/src/util/validate-paths.ts:1]: **convention (non-blocking):** ‏What do you think about using the built-in Node.js version?

```suggestion
import { lstat } from 'fs/promises';
```

File: packages/cli/src/commands/deploy/index.ts [hunk 6]
```diff
@@ -357,6 +345,12 @@ export default async (client: Client): Promise<number> => {
     }
   }
 
+  // For repo-style linking, reset the path to the root of the repository
+  //console.log(link);
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/deploy/index.ts]: **suggestion (soft-blocking):** 
‏
```suggestion
```

**Review by EndangeredMassa — COMMENTED** [review]
Looking good!

File: packages/cli/src/commands/deploy/index.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/cli/src/commands/deploy/index.ts]: You're right, good catch.

```suggestion
    cwd = link.repoRoot;
```

File: packages/cli/src/util/deploy/process-deployment.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/cli/src/util/deploy/process-deployment.ts:73]: Fine by me. Let's do in a separate PR though (would love to refactor out the entire legacy `Now` client).

File: packages/cli/src/util/link/repo.ts (hunk 3 above)
Comment (TooTallNate) [reply packages/cli/src/util/link/repo.ts:287]: I plan on addressing the use-case where there's multiple projects with the same root directory (i.e. Storybook) in the next PR. Thus, the singular project match function will go away.

File: packages/cli/src/util/projects/get-project-by-cwd-or-link.ts (hunk 4 above)
Comment (TooTallNate) [reply packages/cli/src/util/projects/get-project-by-cwd-or-link.ts:29]: Prettier <picture data-single-emoji=":blob_shrug:" title=":blob_shrug:">:blob_shrug:</picture>

File: packages/cli/src/util/validate-paths.ts (hunk 5 above)
Comment (TooTallNate) [reply packages/cli/src/util/validate-paths.ts:1]: I don't think it makes much difference since we're already using `fs-extra` throughout the code base.

**Review by cb1kenobi — APPROVED** [review]
LGTM. Nice work, especially getting all the path stuff cleaned up!

**Pull request merged**